monai/transforms/lazy/dictionary.py

```diff
@@ -0,0 +1,49 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from __future__ import annotations
+
+from monai.data.meta_tensor import MetaTensor
+from monai.transforms.inverse import InvertibleTransform
+from monai.transforms.lazy.functional import apply_pending
+
+
+class ApplyPendingd(InvertibleTransform):
+    """
+    ApplyPendingd can be inserted into a pipeline that is being executed lazily in order
+    to ensure resampling happens before the next transform. When called, it will check the
+    keys specified by `self.keys` and, if the value at that key is a ``MetaTensor`` instance,
+    it will execute all pending transforms on that value, inserting the transformed ``MetaTensor``
+    at that key location.
+
+    See ``Compose`` for a detailed explanation of the lazy resampling feature.
+    """
+
+    def __init__(self, keys):
+        super().__init__()
+        self.keys = keys
+
+    def __call__(self, data, **kwargs):

```

Comment:
all the dictionary based transform don't currently accept `kwargs` in this at this call

Reply:
ok... so what's the current recommended usage of `ApplyPending` and `ApplyPendingd`, we don't support overriding? I thought we'll have the flexibility of different overrides:
````
[Orientation(),  ApplyPending(overrides_kwargs_1), Flip(), Spacing(), ApplyPending(overrides_kwargs_2)] -> [RandRotate()]
```

Reply:
sure... mark this as resolved. (but is there any difficulty we were unaware of in implementing this? I thought it'd be simply invoking `transforms.lazy.functional.apply_pending_transforms`)

Reply:
The Law of Unintended Consequences
I wouldn't want to put a feature out there that we don't understand and that bites us in the medium term. It's a design choice with potentially significant consequences.

Reply:
What's are the potential consequences could you please make a few concrete examples? (Asking because previously this functionality was one of the main motivations for having this PR.)